src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpTransportBindingElement.cs

```diff
@@ -406,6 +413,20 @@ public override T GetProperty<T>(BindingContext context)
             }
         }
 
+        private MruCache<string, HttpClient> EnsureHttpClientCache()
```

Comment:
Maybe I'm missing something up the stack, but I don't think this is thread safe. Does it need to be?

Reply:
Doesn't need to be thread safe. When this class is first passed to ChannelFactory or any other of our classes, we clone it. This means the call path which gets here will be on an instance not accessible outside of our own code. In our code, we won't call the code path which gets to here on multiple threads simultaneously. It will happen sequentially.